### DISCUSSION TITLE
vi_server_for_ui

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/labview_book#76] [Announcements]
# vi_server_for_ui

在程序运行的过程中，程序的用户界面也不可避免会产生变化。最常见的是界面上某些数据的变化，这一般只需要在程序中把新数据传递给控件的接线端即可。在之前的章节中，我们也介绍了如何使用控件的局部变量和属性对界面做更复杂的调整，比如改变控件的位置颜色等。除了控件，界面上的其它部分，比如窗口的尺寸，标题内容，装饰图片等等也都是可以在程序中进行自动调整的。

https://lv.qizhen.xyz/vi_server_for_ui

**Comment by tetsuoyyx** [comment]
阮老师，我按照您在这里的操作，子VI里面还是无法通过引用访问到具体类型（布尔控件），属性节点上也不显示严格字样，想请教一下，这是为什么啊？

**Reply by ruanqizhen** [reply]
参考一下这个过程：![](https://github.com/ruanqizhen/labview_book/blob/main/docs/images_2/z357.gif?raw=true)

**Reply by leilei319yj** [reply]
阮老师，我如果直接按照上面的过程生成也是不工作的的，这个过程的Untitled 1是不是还少了一个传引用的过程，Untitled 2还需要有一个引用的输入管脚这样就可以了，不然运行的时候报错LabVIEW: (Hex 0x41F) Object reference is invalid.

**Reply by ruanqizhen** [reply]
> 阮老师，我如果直接按照上面的过程生成也是不工作的的，这个过程的Untitled 1是不是还少了一个传引用的过程，Untitled 2还需要有一个引用的输入管脚这样就可以了，不然运行的时候报错LabVIEW: (Hex 0x41F) Object reference is invalid.

是的。这个图片是为了回答上面一个读者关于如何创建严格类型的控件的引用的。实现完整的 VI 功能，还需要参考书中文字部分的说明，完成其他的步骤。
